iroh-relay/src/dns/node_info.rs

```diff
@@ -203,9 +205,7 @@ impl NodeInfo {
 /// Takes a [`hickory_resolver::proto::rr::Name`] DNS name and expects the first label to be
 /// [`IROH_TXT_NAME`] and the second label to be a z32 encoded [`NodeId`]. Ignores
 /// subsequent labels.
-pub(crate) fn node_id_from_hickory_name(
-    name: &hickory_resolver::proto::rr::Name,
-) -> Option<NodeId> {
+pub fn node_id_from_hickory_name(name: &hickory_resolver::proto::rr::Name) -> Option<NodeId> {
```

Comment:
Unfortunate that this becomes fully `pub`, it sounds very implementation-y

Reply:
Made it `pub(crate)` and added a `node_id_from_domain_name(domain_name: &str)` pub method instead. Small performance penalty if using that while having a hickory `Name` in a crate using this, but in this way the API is not tied to hickory.